File: mailer/api/src/lib/resend.ts
import { Resend } from 'resend';

export function sendEmail({from , to , subject, html}) {

const resend = new Resend('re_ZsPRRebd_MMf2ArsA9HXEh2yZCu5vY1ed');


resend.emails.send({
  from:  from || 'onboarding@resend.dev',
  to: to ||'admin@mosque.icu',
  subject: subject ||  'Hello World',
  html: html || '<p>Congrats on sending your <strong>first email</strong>!</p>'
});


}

File: api/lib/resend/resend.go
package resend

import (
	"log"

	"github.com/resendlabs/resend-go"
)

func SendEmail(to,from,subject,html) (bool, error) {
    apiKey := "re_Knpe3rhM_ExDXKXY55GTZjafbQom2Mjh2"

    client := resend.NewClient(apiKey)

    params := &resend.SendEmailRequest{
        From:    "onboarding@resend.dev",
        To:      []string{"admin@mosque.icu"},
        Subject: "Hello World",
        Html:    "<p>Congrats on sending your <strong>first email</strong>!</p>",
    }

    sent, err := client.Emails.Send(params)
    if err != nil {
        return false, err
    }
    log.Println(sent)
    return true, err
}
